### PR TITLE
Demo page: Show invalid results on page with details

### LINKED ISSUE #32: Log which filename caused a parse error on the demo page
If parsing errors, the demo page should log which file caused the error. Useful for when you try to import a folder of files.

These errors currently get shown, but aren't really helpful:

`main.js:57 No supported broker found!`

```js
getBroker @ index.js:31
getActivity @ index.js:9
extractActivities @ index.js:72
async function (async)
extractActivities @ index.js:55
fileReader.onload @ main.js:57
load (async)
createActivityFromPDF @ main.js:56
processFiles @ main.js:51
fileHandler @ main.js:48
invokeWithErrorHandling @ vue.js:1863
invoker @ vue.js:2188
original._wrapper @ vue.js:7547
Show 4 more frames
main.js:71 []length: 0__proto__: Array(0)
onvista.js:145 Error while parsing page (onvista RangeError: Invalid time value
    at format (index.js:371)
    at parseData (onvista.js:122)
    at Module.parsePages (onvista.js:142)
    at getActivity (index.js:13)
    at extractActivities (index.js:72)
    at async FileReader.fileReader.onload (main.js:57) (116) [ FILE CONTENTS AS ARRAY ]
parsePages @ onvista.js:145
getActivity @ index.js:13
extractActivities @ index.js:72
async function (async)
extractActivities @ index.js:55
fileReader.onload @ main.js:57
load (async)
createActivityFromPDF @ main.js:56
processFiles @ main.js:51
fileHandler @ main.js:48
invokeWithErrorHandling @ vue.js:1863
invoker @ vue.js:2188
original._wrapper @ vue.js:7547
Show 3 more frames
main.js:71 []
```

### LINKED ISSUE #432: Demo page doesn't output json/csv of not-parsed files
This used to work sometime ago and is also documented in the Readme.md of this repo.

### PR DESCRIPTION
**Pull request by dredav**
- Fix #32
- Fix #432
- Refactoring: Exctract the file parsing from the default export into `parseFile`
Screen Shot 2020-12-22 at 06 22 38

### CODE REVIEW

**Comment by dredav** [comment]
@carstenhag Did you need more information for failed documents at the demo page?

**Comment by carstenhag** [comment]
@dredav um dir das genau sagen zu können bräuchte ich ein paar Minuten um das zu bauen etc. Weiß nicht wann ichs schaffe die Tage.

**Review by SirGibihm — DISMISSED** [review]
looks good! @carstenhag Do you want to look it over as well? I just checked the code and did not test the functionality as I do not have the test .pdf files.

**Comment by dredav** [comment]
@SirGibihm I added a minor change.

**Review by SirGibihm — APPROVED** [review]
Jep, looks good!

**Comment by carstenhag** [comment]
Just checked it out and ran it. First I encountered an issue as I had ran only `npm run build`, but apparently I also need to run `npm run dev` in order for `tresor-demo.js` to be built. Either this is broken or should be documented better I think?

File: demo/index.html [hunk 1]
```diff
@@ -48,59 +48,90 @@ <h1 class="title">
...
-                  <p>
-                    <small class="is-size-7">Tax: {{formatPrice(activity.tax)}}€</small>
+                  <p style="margin-left: .3em">
+                    <small>Anzahl Seiten: {{ error.content.length }}</small>
                   </p>
-                  <p>
-                    <small class="is-size-7">Fee: {{formatPrice(activity.fee)}}€</small>
+                  <p style="margin-left: .3em">
+                    <textarea :id="'content-' + i" style="display: none">{{ error.content }}</textarea>
+                    <small><a href="#" @click="copyContentToClipboard(i)">Inhalt kopieren</a></small>
```
Comment (carstenhag) [inline demo/index.html:79]:
```suggestion
                    <small><a href="#" @click="copyContentToClipboard(i)">JSON-Inhalt kopieren</a></small>
```

**Comment by carstenhag** [comment]
Apart from the npm run issue it looks nice to me, thanks for improving this! 👍

**Comment by dredav** [comment]
>Either this is broken or should be documented better I think?

@carstenhag: The `npm build` command is for the production build process. Yes, you need to run the `npm dev` command to build the local and develop state. But then you also need to run `npm demo`. Instead of `npm dev` and `npm demo` you can use `npm start` which run both commands (See README.md, contribute, 2.: `install and start npm i && npm start`).

**Comment by carstenhag** [comment]
Ahh I see, thanks for explaining. Must have skipped that :)

**Review by SirGibihm — APPROVED** [review]

**Pull request merged**